lib/Doctrine/DBAL/Driver/IBMDB2/DB2Connection.php

```diff
@@ -116,9 +116,23 @@ public function exec(string $statement) : int
     /**
      * {@inheritdoc}
      */
-    public function lastInsertId(?string $name = null) : string
+    public function lastInsertId() : string
     {
-        return db2_last_insert_id($this->conn);
+        $lastInsertId = db2_last_insert_id($this->conn);
+
+        if ($lastInsertId === '') {
+            throw new DB2Exception('No identity value was generated by the last statement.');
+        }
+
+        return $lastInsertId;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSequenceNumber(string $name) : string
+    {
+        throw new DB2Exception('Sequences on IBM DB2 are not currently supported.');
```

Comment:
This does not really imply that DB2 does not support them, but I understand that this can be a source of confusion. Do you prefer:

> Sequences on IBM DB2 are not currently supported **by the DBAL**.

Reply:
Fixed in 3d93734e77b49b01f847bfaf5a0ea3b1159178d5.